options.go

```diff
@@ -76,49 +76,49 @@ type ShrinkThrOptions struct {
 }
 
 // GetShrinkThreadOpts return a copy of ShrinkThrOptions.
-func (o *options) GetShrinkThreadOpts() ShrinkThrOptions {
+func (o *options) GetShrinkThreadOpts() *ShrinkThrOptions {
 	o.L.RLock()
 	defer o.L.RUnlock()
-	return *o.ShrinkThrOptions
+	return o.ShrinkThrOptions
 }
 
-// GetMemOpts return a copy of memOpts.
-func (o *options) GetMemOpts() typeOption {
+// GetMemOpts return a copy of typeOption.
+func (o *options) GetMemOpts() *typeOption {
 	o.L.RLock()
 	defer o.L.RUnlock()
-	return *o.memOpts
+	return o.memOpts
 }
 
-// GetCPUOpts return a copy of cpuOpts
+// GetCPUOpts return a copy of typeOption
 // if cpuOpts not exist return a empty typeOption and false.
-func (o *options) GetCPUOpts() typeOption {
+func (o *options) GetCPUOpts() *typeOption {
 	o.L.RLock()
 	defer o.L.RUnlock()
-	return *o.cpuOpts
+	return o.cpuOpts
```

Comment:
Nope, we should return the full copy before unlock.
returning a point is not safe here, the locking is useless then.

Also, the point should not be nil unless people use holmes unexpectly.

Reply:
Okay, there's something wrong with my understanding.